stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -119,10 +125,15 @@ class EntryPoint:
     def resolve(self) -> Any: ...
 
 def find_distributions(path_item: str, only: bool = ...) -> Generator[Distribution, None, None]: ...
-def get_distribution(dist: Requirement | str | Distribution) -> Distribution: ...
+@overload
+def get_distribution(dist: _D) -> _D: ...
+@overload
+def get_distribution(dist: _PkgReqType) -> DistInfoDistribution: ...
```

Comment:
This calls `get_provider()`, which according to our stub below returns a `Distribution`, not a `DistInfoDistribution`. I haven't traced the code to verify which is right, but we should be consistent.

Reply:
![image](https://user-images.githubusercontent.com/1350584/220843885-99610aa1-44a3-4463-8609-955eccdbfd0a.png)

Reply:
I think you're right. Didn't look too far into it but might depend on if the distribution was `.egg`, `.egg-info` or `.dist-info`